### ISSUE TITLE
fix(main): configureRibbonIcons leaks a ribbon element and a DOM listener per save

### ISSUE THREAD

**Issue by philoserf** [philoserf/obsidian-periodic-notes#193]
**Severity:** low
**Location:** `src/main.ts:75-99`

## Description

```ts
private configureRibbonIcons(): void {
  this.ribbonEl?.detach();
  const granularity = granularities.find(...);
  if (granularity) {
    this.ribbonEl = this.addRibbonIcon(...);
    this.ribbonEl.addEventListener("contextmenu", (e: MouseEvent) => {...});
  }
}
```

`saveSettings` calls this on every save (`main.ts:127`), and saves are frequent
(debounced per keystroke burst in three text fields, plus the un-debounced
Enabled toggle). Each call:

- Calls `addRibbonIcon` again, which creates a new element *and* adds a fresh
  unload callback to the plugin's own cleanup list. `detach()` removes the
  previous element from the DOM but does not undo that registration, so the
  list grows without bound for the lifetime of the plugin.
- Attaches the `contextmenu` handler with a raw `addEventListener` rather than
  `registerDomEvent`, so it is not tracked for cleanup and its closure over
  `this` keeps the detached element reachable.

`this.ribbonEl` is also left pointing at a detached element when no granularity
is enabled — harmless, since `detach()` on an already-detached node is a no-op,
but it means the field no longer describes what is on screen.

This is a hygiene problem rather than a user-visible failure: the ribbon renders
correctly throughout. It is unrelated to #160, which asks for one icon per
enabled granularity.

## Suggested fix

Track and clear the element properly — after `detach()`, set
`this.ribbonEl = null` — and use `this.registerDomEvent(this.ribbonEl, "contextmenu", ...)`
so the listener is torn down with the plugin. Better still, rebuild only when
the leading enabled granularity actually changed: cache it and return early when
it matches, which removes the churn entirely for format/folder/template edits.

---

Source: code-audit `.issues/ribbon-icon-rebuild-leaks-per-save.md`, verified against `808f8c6`.
